src/main/java/com/hypherionmc/modpublisher/tasks/GithubUploadTask.java

```diff
@@ -86,19 +86,22 @@ public void upload() throws Exception {
 
         GHRepository ghRepository = gitHub.getRepository(uploadRepo);
 
+        final String tag = extension.getGithubTag().get();
+
         // Try to find an existing release.
         // If one is found, the file will be added onto it.
-        GHRelease ghRelease = ghRepository.getReleaseByTagName(extension.getVersion().get());
+        GHRelease ghRelease = ghRepository.getReleaseByTagName(tag);
 
         UploadPreChecks.checkEmptyJar(extension, uploadFile, extension.getLoaders().get());
 
         // Existing release was not found, so we create a new one
         if (ghRelease == null) {
-            GHReleaseBuilder releaseBuilder = new GHReleaseBuilder(ghRepository, extension.getVersion().get());
+            GHReleaseBuilder releaseBuilder = new GHReleaseBuilder(ghRepository, tag);
 
             if (extension.getDisplayName().isPresent() && !extension.getDisplayName().get().isEmpty()) {
                 releaseBuilder.name(extension.getDisplayName().get());
             } else {
+                // TODO make version optional if tag is set
```

Comment:
Since tag defaults to version, would it make sense to just pass tag here, instead of hard pulling it from version?

Reply:
IDK. Usually a name would be defined, but if not would a user expect the tag or the version to be used (assuming they're different)?

Reply:
The version fallback was just added here, since afaik, GitHub does require a name for releases? I could be wrong about this.

Maybe it can be a case of using the display name, or tag if it differs from the version. Idk, or otherwise this should be another config setting.

Reply:
> Maybe it can be a case of using the display name, or tag if it differs from the version

Well since `tag` defaults to `version`, it will always be defined.

`version` could be in any of three states; undefined, matching, or different.

The main reason for `tag` to exist in the first place is to allow `version` to be different, but if a user only cares about GitHub and isn't publishing to Modrinth, `version` could be optional...

My `TODO` comment had something like `version().getOrDefault(tag)` in mind. Does that work for you?

Reply:
> version().getOrDefault(tag)
This sounds good to me. Plain and simple

Reply:
Done